framework/include/auxkernels/AuxKernel.h

```diff
@@ -126,22 +126,26 @@ class AuxKernel :
   SubProblem & _subproblem;
   /// System this kernel is part of
   SystemBase & _sys;
+  /// Nonlinear system, displaced if this kernel is on the auxiliary displaced system
   SystemBase & _nl_sys;
+  /// Auxiliary undisplaced system
   AuxiliarySystem & _aux_sys;
   /// Thread ID
   THREAD_ID _tid;
   /// Assembly class
   Assembly & _assembly;
   /// Variable this kernel is acting on
   MooseVariable & _var;
+  /// Displaced problem (valid only when it exists)
+  MooseSharedPointer<DisplacedProblem> _displaced_problem;
```

Comment:
:)

On Fri, Jun 3, 2016 at 6:47 PM Derek Gaston <notifications@github.com>
wrote:

> In framework/include/auxkernels/AuxKernel.h
> <https://github.com/idaholab/moose/pull/7151#discussion_r65793734>:
>
> >    AuxiliarySystem & _aux_sys;
> >    /// Thread ID
> >    THREAD_ID _tid;
> >    /// Assembly class
> >    Assembly & _assembly;
> >    /// Variable this kernel is acting on
> >    MooseVariable & _var;
> > +  /// Displaced problem (valid only when it exists)
> > +  MooseSharedPointer<DisplacedProblem> _displaced_problem;
>
> @permcody <https://github.com/permcody> I'm starting to see the problem
> here. It's a bad idea for objects like AuxKernels to have shared pointers
> to objects like the Systems. It doesn't make sense. AuxKernel doesn't have
> shared ownership of the DisplacedProblem...
>
> @YaqiWang <https://github.com/YaqiWang> I'm not saying anything against
> this code... you're doing the "right" thing... the interface returns a
> shared pointer and that's how you're catching it...
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7151/files/63ee705db72c0e8b66427760050fbdd412f0af59#r65793734>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC5XIMyfBRB8oRe4ucLshI5UNiqX0Xw7ks5qIMsLgaJpZM4ItxmV>
> .
>